text/0119-standard-system-metrics.md

```diff
@@ -0,0 +1,150 @@
+
+#### OS Specific System Metrics - `system.{os}.`
+
+Instrument names for system level metrics specific to a certain operating system should be prefixed with `system.{os}.` and follow the hierarchies listed above for different entities like CPU, memory, and network. For example, an instrument for counting the number of Linux merged disk operations (see [here](https://unix.stackexchange.com/questions/462704/iostat-what-is-exactly-the-concept-of-merge) and [here](https://man7.org/linux/man-pages/man1/iostat.1.html)) could be named `system.linux.disk.merged_ops`, reusing the `disk` name proposed above.
+
+### Standard Runtime Metrics - `runtime.`
+
+---
+
+Runtime environments vary widely in their terminology, implementation, and relative values for a given metric. For example, Go and Python are both garbage collected languages, but comparing heap usage between the two runtimes directly is not meaningful. For this reason, this OTEP does not propose any standard top-level runtime metric instruments. See [OTEP #108](https://github.com/open-telemetry/oteps/pull/108/files) for additional discussion.
+
+#### Runtime Specific Metrics - `runtime.{environment}.`
+
+Runtime level metrics specific to a certain runtime environment should be prefixed with `runtime.{environment}.` and follow the semantic conventions outlined in [Semantic Conventions](#semantic-conventions). For example, Go runtime metrics use `runtime.go.` as a prefix.
+
+Some programming languages have multiple runtime environments that vary significantly in their implementation, for example [Python has many implementations](https://wiki.python.org/moin/PythonImplementations). For these languages, consider using specific `environment` prefixes to avoid ambiguity, like `runtime.cpython.` and `runtime.pypy.`.
+
+## Open questions
+
+- Should the individual runtimes have their specific naming conventions in the spec?
+- Is it ok to include instruments specific to an OS (or OS family) under a top-level prefix, as long as they are unambiguous? For example, naming inode related instruments, which of the below is preferred?
+  1. Top level: `system.filesystem.inodes.*`
+  2. UNIX family level: `system.unix.filesystem.inodes.*`
+  3. One for each UNIX OS: `system.linux.filesystem.inodes.*`, `system.freebsd.filesystem.inodes.*`, `system.netbsd.filesystem.inodes.*`, etc.
+- Should instrument names be versioned e.g. `v1.system.cpu.usage`?
+  * Some telemetry backends use a strict schema for metrics and their labels (e.g. Google Cloud Monitoring with [MetricDescriptors](https://cloud.google.com/monitoring/api/v3/metrics-details#intro-metdesc)), so changes to the instruments proposed in this document could break ingestion of new telemetry data.
+  * Should the exporters for such backends be responsible for handling this? How could this be implemented in the exporter without additional information?
```

Comment:
I opened a spec issue for this, it looks like a strong "no". I will remove this and update

Reply:
The use of strict schemas is a very Prometheus-ey thing, not common in the Statsd world. I would prefer if systems could perform some aggregation to transform data into the schema they want as opposed to not recognizing matching time series. For example: your system expects to monitor a system metric w/ four labels, but it arrives with 5 labels? You can aggregate away one label and get the result you need. In any case, it doesn't sound like this issue is blocking this OTEP.

Reply:
Here's the relevant issue https://github.com/open-telemetry/opentelemetry-specification/issues/703